### PR TITLE
fix: Improves assignment activity for #186.

### PR DESCRIPTION
**Pull request by ryasmi**
Going to need a test from @lydiamross on Moodle.

- `\mod_assign\event\assessable_submitted`
- `\mod_assign\event\submission_graded`

Part of the solution to #186.

### CODE REVIEW

File: src/transformer/events/mod_assign/assignment_submitted.php [hunk 1]
```diff
@@ -34,7 +36,7 @@ function assignment_submitted(array $config, \stdClass $event) {
                 $lang => 'submitted'
             ],
         ],
-        'object' => utils\get_activity\event_module($config, $event, $lang),
+        'object' => utils\get_activity\course_assignment($config, $event->contextinstanceid, $assignment->name, $lang),
         'timestamp' => utils\get_event_timestamp($event),
         'context' => [
```
Comment (lydiamross) [inline src/transformer/events/mod_assign/assignment_submitted.php:41]: @ryansmith94 
The displayed xapi statement has erroneous data in context.extensions -
[completedAssignment.txt](https://github.com/xAPI-vle/moodle-logstore_xapi/files/2249025/completedAssignment.txt)

**Review by lydiamross — COMMENTED** [review]
\mod_assign\event\submission_graded is fine

\mod_assign\event\assessable_submitted needs a look

File: src/transformer/events/mod_assign/assignment_submitted.php (hunk 1 above)
Comment (ryasmi) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: What is the erroneous data @lydiamross?

Comment (ryasmi) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: I can't see what's erroneous in the context extensions there

Comment (lydiamross) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: Line 35-36 ?

Comment (lydiamross) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: Line 35-36 in the attached statement, I mean

Comment (ryasmi) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: This bit here?
```
"event_name": "\\core\\event\\course_module_completion_updated",
"event_function": "\\src\\transformer\\events\\core\\course_module_completion_updated"
```

That looks like a different event, is it possible that the `\mod_assign\event\assessable_submitted` just isn't being sent, but it also triggers the completion event in the background? Perhaps check if `\core\event\course_module_completion_updated` appears in your standard log straight after/before `\mod_assign\event\assessable_submitted`.

Comment (lydiamross) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: Or is 

"event_name": "\\core\\event\\course_module_completion_updated",
"event_function": "\\src\\transformer\\events\\core\\course_module_completion_updated"

intentional?

Comment (lydiamross) [reply src/transformer/events/mod_assign/assignment_submitted.php]: This seems ok now

Comment (ryasmi) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: Definitely not intentional for `\mod_assign\event\assessable_submitted`

Comment (lydiamross) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: You're right - I was thinking \core\event\course_module_completion_updated meant something else. That means that \mod_assign\event\assessable_submitted is not being sent to the lrs.

Comment (ryasmi) [reply src/transformer/events/mod_assign/assignment_submitted.php:41]: Okay we'll figure out this one tomorrow.

**Review by lydiamross — APPROVED** [review]

**Pull request merged**